### PR TITLE
week3-homework-omer-kilic

### PR DESCRIPTION
**Pull request by obeka**
To view normalization.md, please click here : https://github.com/obeka/databases/blob/week3-homework-omer-kilic/Week3/homework/normalization.md

For this part of the hw :
_*Write down all the steps of conversion (installation, commands etc.) in a text file / MD file._
I prepared a video before explaining migration from mysql to mongodb. Here is the link: 
https://youtu.be/H9PlhBSzysA

### CODE REVIEW

**Comment by obeka** [comment]
To view normalization.md , please click here: 

> For this part of the hw :
> _*Write down all the steps of conversion (installation, commands etc.) in a text file / MD file._
> I prepared a video before explaining migration from mysql to mongodb. Here is the link:
> https://youtu.be/H9PlhBSzysA

File: Week3/homework/normalization.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                | dinner_id | member_id |
+                +-----------+-----------+
+                | D00001001 | 1         |
+                | D00001002 | 2         |
+                | D00001002 | 3         |
+                | D00001003 | 4         |
+                | D00001003 | 8         |
+                | D00001004 | 6         |
+                | D00001005 | 7         |
+                +-----------+-----------+
```
Comment (fversnel) [inline Week3/homework/normalization.md:130]: Very well put! I like the drawings, they make things very clear!

File: Week3/homework/sql-injection/sql-injection.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    password: 'hyfpassword',
+    database: 'world',
+    multipleStatements: true
+});
+
+conn.connect();
+
+function getPopulation(Country, name, code, cb) {
+    conn.query(
+        `SELECT Population FROM ${Country} WHERE Name = '${name}' and code = ${code}`,
```
Comment (fversnel) [inline Week3/homework/sql-injection/sql-injection.js:16]: Possible SQL injection attack. Please look up how to protect against an attack: https://github.com/mysqljs/mysql#escaping-query-values

File: Week3/homework/transaction/tr-create-tables.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        amount INT,
+        changed_date DATETIME,
+        remark VARCHAR(50) 
+    );
+    `;
+
+    connection.connect();
+
+  try {
+    await Promise.all[execQuery(CREATE_A_DATABASE),execQuery(SELECT_DB),execQuery(CREATE_ACCOUNT_TABLE),execQuery(CREATE_ACCOUNT_CHANGES_TABLE)];
```
Comment (fversnel) [inline Week3/homework/transaction/tr-create-tables.js:42]: There's quite a bit happening on this line and you're mixing promises with async await which are two different abstractions. Could you do it for using a regular for loop and using just async/await?
You're mixing the order of the operations together with the execution but what if you started with:
`const operations = [CREATE_DATABASE, SELECT_DB, ...]` and continued from there?

File: Week3/homework/mongoDB/mongodb.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+    console.log(`${result.matchedCount} document(s) matched the query criteria.`);
+    console.log(`${result.modifiedCount} document(s) was/were updated.`);
+}
+
+async function deleteCityByName(db, cityName) {
+    result = await db.collection("city")
+            .deleteOne({ name: cityName });
+    console.log(`${result.deletedCount} document(s) was/were deleted.`);
+}
```
Comment (fversnel) [inline Week3/homework/mongoDB/mongodb.js:86]: I'm not the world's biggest mongo expert. As far as I can see it looks good.

File: Week3/homework/sql-injection/sql-injection.js (hunk 2 above)
Comment (obeka) [reply Week3/homework/sql-injection/sql-injection.js:16]: Hi, @fversnel, this is the line which is vulnurable for SQL injection attack, after line 31, secure function is located. I want to put them both in my js file to see the differences.

File: Week3/homework/transaction/tr-create-tables.js (hunk 3 above)
Comment (obeka) [reply Week3/homework/transaction/tr-create-tables.js]: > Very well done homework, just one minor point in the comments :)

Hi @fversnel, Thanks for your overall review. I pushed my change on your comment, hope it is what you want. Look to hear from you again.

**Comment by Tjebbee** [comment]
hey @fversnel , can this homework be approved?

**Comment by fversnel** [comment]
Yep, it's good :)

**Pull request closed without merging**